worlds/sm64ex/Items.py

```diff
@@ -1,47 +1,59 @@
 
-action_item_table = {
-    "Double Jump": 3626185,
-    "Triple Jump": 3626186,
-    "Long Jump": 3626187,
-    "Backflip": 3626188,
-    "Side Flip": 3626189,
-    "Wall Kick": 3626190,
-    "Dive": 3626191,
-    "Ground Pound": 3626192,
-    "Kick": 3626193,
-    "Climb": 3626194,
-    "Ledge Grab": 3626195
+action_item_data_table: dict[str, SM64ItemData] = {
+    "Double Jump": SM64ItemData(sm64ex_base_id + 185, ItemClassification.progression),
+    "Triple Jump": SM64ItemData(sm64ex_base_id + 186, ItemClassification.progression),
+    "Long Jump": SM64ItemData(sm64ex_base_id + 187, ItemClassification.progression),
+    "Backflip": SM64ItemData(sm64ex_base_id + 188, ItemClassification.progression),
+    "Side Flip": SM64ItemData(sm64ex_base_id + 189, ItemClassification.progression),
+    "Wall Kick": SM64ItemData(sm64ex_base_id + 190, ItemClassification.progression),
+    "Dive": SM64ItemData(sm64ex_base_id + 191, ItemClassification.progression),
+    "Ground Pound": SM64ItemData(sm64ex_base_id + 192, ItemClassification.progression),
+    "Kick": SM64ItemData(sm64ex_base_id + 193, ItemClassification.progression),
+    "Climb": SM64ItemData(sm64ex_base_id + 194, ItemClassification.progression),
+    "Ledge Grab": SM64ItemData(sm64ex_base_id + 195, ItemClassification.progression),
 }
 
+cannon_item_data_table: dict[str, SM64ItemData] = {
+    "Cannon Unlock BoB": SM64ItemData(sm64ex_base_id + 200, ItemClassification.progression),
+    "Cannon Unlock WF": SM64ItemData(sm64ex_base_id + 201, ItemClassification.progression),
+    "Cannon Unlock JRB": SM64ItemData(sm64ex_base_id + 202, ItemClassification.progression),
+    "Cannon Unlock CCM": SM64ItemData(sm64ex_base_id + 203, ItemClassification.progression),
+    "Cannon Unlock SSL": SM64ItemData(sm64ex_base_id + 207, ItemClassification.progression),
+    "Cannon Unlock SL": SM64ItemData(sm64ex_base_id + 209, ItemClassification.progression),
+    "Cannon Unlock WDW": SM64ItemData(sm64ex_base_id + 210, ItemClassification.progression),
+    "Cannon Unlock TTM": SM64ItemData(sm64ex_base_id + 211, ItemClassification.progression),
+    "Cannon Unlock THI": SM64ItemData(sm64ex_base_id + 212, ItemClassification.progression),
+    "Cannon Unlock RR": SM64ItemData(sm64ex_base_id + 214, ItemClassification.progression),
+}
 
-cannon_item_table = {
-    "Cannon Unlock BoB": 3626200,
-    "Cannon Unlock WF": 3626201,
-    "Cannon Unlock JRB": 3626202,
-    "Cannon Unlock CCM": 3626203,
-    "Cannon Unlock SSL": 3626207,
-    "Cannon Unlock SL": 3626209,
-    "Cannon Unlock WDW": 3626210,
-    "Cannon Unlock TTM": 3626211,
-    "Cannon Unlock THI": 3626212,
-    "Cannon Unlock RR": 3626214
+item_data_table = {
+    **generic_item_data_table,
+    **action_item_data_table,
+    **cannon_item_data_table
 }
 
-item_table = {**generic_item_table, **action_item_table, **cannon_item_table}
\ No newline at end of file
+item_table = {name: data.code for name, data in item_data_table.items() if data.code is not None}
```

Comment:
Might as well add the EoF newline now
```suggestion
item_table = {name: data.code for name, data in item_data_table.items() if data.code is not None}

```